components/ui-api-layer/internal/domain/k8s/replicaset_service_test.go

```diff
@@ -0,0 +1,329 @@
+		require.Error(t, err)
+		assert.True(t, errors.IsInvalid(err))
+		assert.Nil(t, replicaSet)
+
+		replicaSet, err = client.ReplicaSets(exampleNamespace).Get(exampleName, metav1.GetOptions{})
+		require.NoError(t, err)
+		assert.Equal(t, exampleReplicaSet, replicaSet)
+	})
+
+	t.Run("InvalidUpdate", func(t *testing.T) {
+		exampleName := "exampleReplicaSet"
+		exampleNamespace := "exampleNamespace"
+		exampleReplicaSet := fixReplicaSet(exampleName, exampleNamespace, nil)
+		replicaSetInformer, client := fixFailingReplicaSetInformer(exampleReplicaSet)
+		svc := k8s.NewReplicaSetService(replicaSetInformer, client)
+
+		update := exampleReplicaSet.DeepCopy()
+		update.Labels = map[string]string{
+			"example": "example",
+		}
+
+		replicaSet, err := svc.Update(exampleName, exampleNamespace, *update)
+		require.Error(t, err)
+		assert.Nil(t, replicaSet)
+
+		replicaSet, err = client.ReplicaSets(exampleNamespace).Get(exampleName, metav1.GetOptions{})
+		require.NoError(t, err)
+		assert.Equal(t, exampleReplicaSet, replicaSet)
+	})
+
+	t.Run("TypeMetaChanged", func(t *testing.T) {
+		exampleName := "exampleReplicaSet"
+		exampleNamespace := "exampleNamespace"
+		exampleReplicaSet := fixReplicaSet(exampleName, exampleNamespace, nil)
+		replicaSetInformer, client := fixReplicaSetInformer(exampleReplicaSet)
+		svc := k8s.NewReplicaSetService(replicaSetInformer, client)
+
+		update := exampleReplicaSet.DeepCopy()
+		update.Kind = "OtherKind"
+		replicaSet, err := svc.Update(exampleName, exampleNamespace, *update)
+		require.Error(t, err)
+		assert.True(t, errors.IsInvalid(err))
+		assert.Nil(t, replicaSet)
+
+		update.Kind = "ReplicaSet"
+		update.APIVersion = "OtherVersion"
+		replicaSet, err = svc.Update(exampleName, exampleNamespace, *update)
+		require.Error(t, err)
+		assert.True(t, errors.IsInvalid(err))
+		assert.Nil(t, replicaSet)
+
+		replicaSet, err = client.ReplicaSets(exampleNamespace).Get(exampleName, metav1.GetOptions{})
+		require.NoError(t, err)
+		assert.Equal(t, exampleReplicaSet, replicaSet)
+	})
+
+	t.Run("NoTypeMetaReturned", func(t *testing.T) {
+		exampleName := "exampleReplicaSet"
+		exampleNamespace := "exampleNamespace"
+		returnedReplicaSet := fixReplicaSetWithoutTypeMeta(exampleName, exampleNamespace, nil)
+		expectedReplicaSet := fixReplicaSet(exampleName, exampleNamespace, nil)
+		replicaSetInformer, client := fixReplicaSetInformer(returnedReplicaSet)
+		svc := k8s.NewReplicaSetService(replicaSetInformer, client)
+
+		update := expectedReplicaSet.DeepCopy()
+		update.Labels = map[string]string{
+			"example": "example",
+		}
+
+		replicaSet, err := svc.Update(exampleName, exampleNamespace, *update)
+		require.NoError(t, err)
+		assert.Equal(t, update, replicaSet)
+
+		replicaSet, err = client.ReplicaSets(exampleNamespace).Get(exampleName, metav1.GetOptions{})
+		require.NoError(t, err)
+		assert.Equal(t, update, replicaSet)
+	})
+}
+
+func TestReplicaSetService_Delete(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
```

Comment:
It would be good to test also error case. It would be easy with method `PrependReactor` - please search for usages in the component directory.